### DISCUSSION TITLE
Output as Python?

### DISCUSSION THREAD

**Discussion by Captain-Bacon** [langflow-ai/langflow#182] [Ideas]
I was wondering if it was possible to parse the output as python script, as if it were written for langchain 'natively' in python?

I ask because I want to play around using some compression techniques for prompting, and maybe some other stuff, and at the moment I don't think this is possible?

**Comment by ngmisl** [comment]
relevant question. I've asked this in the webinar earlier this week, and assumed there would be some kind of option?

**Reply by ogabrielluiz** [reply]
Do you mean having a script that instantiates each variable as if it were the graph in LangFlow?

We do have the option to use the built flow(through langflow.load_flow_from_json) and have also plans to improve the prompting experience inside LangFlow but exporting the code itself is not on our roadmap yet.

We have plans on improving the developer experience for debugging flows in python but this does not include editing the nodes yet (even though it is possible).

**Reply by Captain-Bacon** [reply]
I’m really sorry, but I’m too ignorant to know the terminology.  I think what I'm asking is that instead of a JSON file,  if there could be the 'full' python code for running.  Like, everything.

for example if I create a chain of some type I then export as complete code.  I can then import/paste into an IDE like VS Code, and run it from scratch.  It would look and feel like I had written the python from the ground up.  All the imports, defs, variables etc. 

It would allow someone like me to get a better understanding via implementation - I know what it's doing, I can reverse engineer my way through it to get an understanding of it, and then I can add things that maybe Langflow can't do, or experiment.

**Reply by ngmisl** [reply]
exactly. use langflow to prototype and then take the full python script for using outside

**Comment by ogabrielluiz** [comment]
There are some things that would make it a bit wonky. 

Specifically, the PythonFunction node allows you to add import statements in the code, and having it in the middle of the script would be kind of weird. 

The webinar the other day got me thinking and prototyping some of this but we got some things that are higher priority atm. 

Would inspecting the graph object be close to your use case?

We are adding a parameter in the `langflow.load_flow_from_json` function that makes it return the graph instead of the object, essentially stopping one step before the default behavior.  This allows you to explore the nodes and edit them if need to.

**Reply by ngmisl** [reply]
Could just run isort on export of the file to avoid issues with imports I guess

**Reply by ogabrielluiz** [reply]
Yes. That is definitely the solution for the imports part.

**Reply by ogabrielluiz** [reply]
Since we build the graph in the order the variables are used, we could just adapt the way we build/import each node to add a line of code to a variable in the graph or just in the node itself then we concatenate everything and validate it somehow.
